Automattic/jetpack
# VideoPress: request purchases to verify if the VideoPress product is included on the site

Fixes https://github.com/Automattic/jetpack/issues/26602

#### Changes proposed in this Pull Request:

* VideoPress:
  * add selectors and resolvers to get purchases data
  * reduce purchases data into its own reducer
  * show/hide upsell depending on whether the VideoPress purchase 

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?



#### Testing instructions:







* Test with a fresh JN site. [Create JN](https://jurassic.ninja/create/?jetpack-beta&branches.jetpack-videopress=update/videopress-get-site-purchases&wp-debug-log).
* Connect the site without buying a product/plan
* Go to VideoPress dashboard
* Confirm you see the upsell nudge

<img width="701" alt="image" src="https://user-images.githubusercontent.com/77539/193950128-58ece6d0-d78e-4a4d-9958-96e652ca4dcd.png">

* you may like to take a look at the async request

<img width="481" alt="Screen Shot 2022-10-04 at 20 43 35" src="https://user-images.githubusercontent.com/77539/193950190-31855a25-4096-4124-b997-0f4533368b0c.png">

* or at the state

<img width="430" alt="Screen Shot 2022-10-04 at 20 44 40" src="https://user-images.githubusercontent.com/77539/193950280-95b91a14-e9e0-4a66-9de0-358e6c7a0715.png">

* Buy a product/plan that includes the VideoPress feature
* Confirm you don't see the upsell nudge anymore
* State should be populated with at least the VideoPress purchase

<img width="755" alt="Screen Shot 2022-10-04 at 20 47 51" src="https://user-images.githubusercontent.com/77539/193950611-87bfbe3d-2436-49c9-b387-54a1d8a4d458.png">
